src/api/app/models/scm_webhook.rb

```diff
@@ -112,11 +116,20 @@ def ignored_github_pull_request_action?
     github_pull_request? && ALLOWED_PULL_REQUEST_ACTIONS.exclude?(@payload[:action])
   end
 
+  def ignored_github_push_event?
+    github_push_event? && @payload[:deleted]
+  end
+
   def ignored_gitlab_merge_request_action?
     gitlab_merge_request? && ALLOWED_MERGE_REQUEST_ACTIONS.exclude?(@payload[:action])
   end
 
   def ignored_gitea_pull_request_action?
     gitea_pull_request? && ALLOWED_PULL_REQUEST_ACTIONS.exclude?(@payload[:action])
   end
+
+  def ignored_gitlab_push_event?
+    # In Push Hook events to delete a branch, the after field is '0000000000000000000000000000000000000000'
+    gitlab_push_event? && @payload[:commit_sha] == '0000000000000000000000000000000000000000' # '0' * 40
```

Comment:
I just thought that this comparison is a bit risky, if the amount of zeros is just off by one it will return someting else

Reply:
sure, was more thinking about our zeros :D but yes, we code this only once, so I'm fine with it